crates/ty_python_semantic/src/module_resolver/resolver.rs

```diff
@@ -47,8 +47,33 @@ pub fn resolve_real_module<'db>(db: &'db dyn Db, module_name: &ModuleName) -> Op
     resolve_module_query(db, interned_name)
 }
 
+/// Resolves a module name to a module (stubs not allowed, some shadowing is
+/// allowed).
+///
+/// In particular, this allows `types` and `typing_extensions` to be
```

Comment:
hmm, `types` _is_ actually a stdlib module, so it's a bit different from `typing_extensions`, which is a "fake typeshed stdlib module"